repos/system_upgrade/el8toel9/actors/checkifcfg/tests/unit_test_ifcfg.py

```diff
@@ -0,0 +1,147 @@
+    ifcfg.process()
+    assert not reporting.create_report.called
+    assert api.produce.called
+    assert isinstance(api.produce.model_instances[0], RpmTransactionTasks)
+    assert api.produce.model_instances[0].to_install == ['NetworkManager-dispatcher-routing-rules']
+
+
+def test_ifcfg_good_type(monkeypatch):
+    """
+    No report if there's an ifcfg file that would work with NetworkManager.
+    Make sure NetworkManager itself is installed though.
+    """
+
+    mock_config = mock.mock_open(read_data="TYPE=Ethernet")
+    with mock.patch("builtins.open" if six.PY3 else "__builtin__.open", mock_config) as mock_ifcfg:
+        monkeypatch.setattr(ifcfg.api, 'current_actor', INITSCRIPTS_AND_NM_INSTALLED)
+        monkeypatch.setattr(ifcfg.api, "produce", produce_mocked())
+        monkeypatch.setattr(ifcfg.os, 'listdir', lambda _: ('hello', 'world', 'ifcfg-eth0', 'ifcfg-lo',))
+        monkeypatch.setattr(ifcfg.os.path, 'isfile', lambda _: True)
+        monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+        ifcfg.process()
+        mock_ifcfg.assert_called_once_with('/etc/sysconfig/network-scripts/ifcfg-eth0')
+        assert not reporting.create_report.called
+        assert api.produce.called
+        assert isinstance(api.produce.model_instances[0], RpmTransactionTasks)
+        assert api.produce.model_instances[0].to_install == ['NetworkManager']
+
+
+def test_ifcfg_not_controlled(monkeypatch):
+    """
+    Report if there's a NM_CONTROLLED=no file.
+    """
+
+    mock_config = mock.mock_open(read_data="TYPE=Ethernet\nNM_CONTROLLED=no")
+    with mock.patch("builtins.open" if six.PY3 else "__builtin__.open", mock_config) as mock_ifcfg:
+        monkeypatch.setattr(ifcfg.api, 'current_actor', INITSCRIPTS_INSTALLED)
+        monkeypatch.setattr(ifcfg.api, "produce", produce_mocked())
+        monkeypatch.setattr(ifcfg.os, 'listdir', lambda _: ('hello', 'world', 'ifcfg-eth0',))
+        monkeypatch.setattr(ifcfg.os.path, 'isfile', lambda _: True)
+        monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+        ifcfg.process()
+        mock_ifcfg.assert_called_once_with('/etc/sysconfig/network-scripts/ifcfg-eth0')
+        assert reporting.create_report.called
+        assert 'disabled NetworkManager' in reporting.create_report.report_fields['title']
+        assert api.produce.called
+
+
+def test_ifcfg_unknown_type(monkeypatch):
+    """
+    Report if there's configuration for a type we don't recognize.
+    """
+
+    mock_config = mock.mock_open(read_data="TYPE=AvianCarrier")
+    with mock.patch("builtins.open" if six.PY3 else "__builtin__.open", mock_config) as mock_ifcfg:
+        monkeypatch.setattr(ifcfg.api, 'current_actor', INITSCRIPTS_AND_NM_INSTALLED)
+        monkeypatch.setattr(ifcfg.api, "produce", produce_mocked())
+        monkeypatch.setattr(ifcfg.os, 'listdir', lambda _: ('hello', 'world', 'ifcfg-pigeon0',))
```

Comment:
please do not use `_` varname anywhere. Use `dummy` instead

Reply:
Is it ok to use `unused` instead of `dummy`? `dummy` is not recommended because it can also be used as an insult.

Reply:
unused is good as well if it's accepted by linters